File: programming files/rgb_to_hex.py
# introduced bugs: ryan langlois

def rgb_to_hex(r, g, b):
    r = max(0, min(255, r))
    g = max(0, min(255, g))
    b = max(0, min(255, b))
    return '{:01X}{:01X}{:01X}'.format(r, g, b)


# test with hex_color = rgb_to_hex(255, 127, 0) # returns "FF7F00"

# this is a comment

File: programming files/rgb_to_hex.cc
#include <iostream>
#include <sstream>
#include <iomanip>

std::string rgb_to_hex(int r, int g, int b) // takes in rgb values
{
    r = std::max(0, std::min(255, r)); // bounds rgb values from 0 to 255
    g = std::max(0, std::min(255, g));
    b = std::max(0, std::min(255, b));

    std::stringstream ss;
    ss << std::uppercase << std::hex << std::setfill('0') // converts rgb to hex
       << std::setw(2) << r << std::setw(2) << g << std::setw(2) << b;

    return ss.str(); // returns the hex value as a stringstream
}

//Test with std::string hexColor = rgb_to_hex(255, 127, 0); // returns "FF7F00"
